cgen error when compiling a program using sumtypes of functions (casting a function value of a variant type to the sumtype, generates invalid code)

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 20.04.3 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz
CC version: cc (Ubuntu 10.3.0-1ubuntu1~20.04) 10.3.0

getwd: /v/vnew
vmodules: /home/delian/.vmodules
vroot: /v/vnew
vexe: /v/vnew/v
vexe mtime: 2022-09-11 11:54:55
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.1 1738641.3e599a1

Git version: git version 2.25.1
Git vroot status: weekly.2022.36-27-g3e599a14 (1 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: stable-thirdparty-linux-amd64 827f7452

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg s.v`
```v
import v.ast

type MyFnSumtype = fn () | fn (int) ast.Expr

fn f(z int) ast.Expr {
	return ast.empty_expr
}

dump( MyFnSumtype(f) )

```

**What did you expect to see?**

A compiled program that prints something like `MyFnSumtype(fn(int)astExpr)`


**What did you see instead?**
```
==================
/tmp/v_1000/s.12928724887323720014.tmp.c:8672: error: ',' expected (got "x")
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/ast/table.v
+++ b/vlib/v/ast/table.v
@@ -230,11 +230,9 @@ pub fn (t &Table) fn_type_signature(f &Fn) string {
 	if f.return_type != 0 && f.return_type != void_type {
 		sym := t.sym(f.return_type)
 		opt := if f.return_type.has_flag(.optional) { 'option_' } else { '' }
-		if sym.kind == .alias {
-			sig += '__$opt$sym.cname'
-		} else {
-			sig += '__$opt$sym.kind'
-		}
+		res := if f.return_type.has_flag(.result) { 'result_' } else { '' }
+
+		sig += '__$opt$res$sym.cname'
 	}
 	return sig
 }

--- a/vlib/v/tests/inout/dump_sumtype_of_fntype.out
+++ b/vlib/v/tests/inout/dump_sumtype_of_fntype.out
@@ -1,0 +1,1 @@
+[vlib/v/tests/inout/dump_sumtype_of_fntype.vv:10] main.MyFnSumtype(main.f): MyFnSumtype(fn (int) v.ast.Expr)

--- a/vlib/v/tests/inout/dump_sumtype_of_fntype.vv
+++ b/vlib/v/tests/inout/dump_sumtype_of_fntype.vv
@@ -1,0 +1,11 @@
+import v.ast
+
+type MyFnSumtype = fn () | fn (int) ast.Expr
+
+fn f(z int) ast.Expr {
+	return ast.empty_expr
+}
+
+fn main() {
+	dump(MyFnSumtype(f))
+}

